overlord/servicestate/quota_handlers.go

```diff
@@ -222,6 +220,86 @@ func addRestartServicesTasks(st *state.State, queueTask func(task *state.Task),
 	}
 }
 
+func (m *ServiceManager) doQuotaAddSnap(t *state.Task, _ *tomb.Tomb) error {
+	st := t.State()
+	st.Lock()
+	defer st.Unlock()
+
+	perfTimings := state.TimingsForTask(t)
+	defer perfTimings.Save(st)
+
+	var quotaName string
+	var snapName string
+	if err := t.Get("snap-name", &snapName); err != nil {
+		return fmt.Errorf("internal error: cannot get snap-name: %v", err)
+	}
+	if err := t.Get("quota-name", &quotaName); err != nil {
+		return fmt.Errorf("internal error: cannot get quota-name: %v", err)
+	}
+
+	if err := CheckQuotaChangeConflictMany(st, []string{quotaName}); err != nil {
+		return err
+	}
+	if err := snapstate.CheckChangeConflictMany(st, []string{snapName}, ""); err != nil {
+		return err
+	}
```

Comment:
conflict checks are not something we do in task handlers, is something we need to do when we create the change

Reply:
Fixed!